reporting-app/app/models/ex_parte_activity.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# ExParteActivity stores hours data submitted via API or batch upload.
+#
+# These are "ex parte" (automated/external) hours as opposed to manually
+# reported hours from ActivityReportApplicationForm. Ex parte hours are
+# auto-verified and don't require staff review.
+#
+# @note The `outside_period` flag should be set by the service layer
+#   (ExParteActivityService) when creating entries, not by callbacks.
+#
+class ExParteActivity < ApplicationRecord
+  ALLOWED_CATEGORIES = %w[employment community_service education].freeze
+
+  SOURCE_TYPE_API = "api"
+  SOURCE_TYPE_BATCH = "batch_upload"
+  ALLOWED_SOURCE_TYPES = [ SOURCE_TYPE_API, SOURCE_TYPE_BATCH ].freeze
```

Comment:
minor nit can we group these to DRY it up a bit
```suggestion
  SOURCE_TYPE = {
    API: "api",
    SOURCE_TYPE_BATCH: "batch_upload"
  }
  ALLOWED_SOURCE_TYPES = SOURCE_TYPE.values.freeze
```